### DISCUSSION TITLE
How to get a callback when a column is updated by any connection?

### DISCUSSION THREAD

**Discussion by pankajkumar229** [sqlalchemy/sqlalchemy#8158] [Usage Questions]
Is it possible in SQL Alchemy to get an event when a row in a DB is updated by other connections (may not be ORM)? The event system seems to act only on ORM updates on current engine(I might be wrong). WE can create a trigger but is it possible to create a callback everytime a trigger is raised?

**Comment by zzzeek** [comment]
SQLAlchemy can only have events for the SQL that is originated from the Python code in terms of  a particular engine or engines (at the Core level, not the ORM), but there's no system in SQLAlchemy to listen for general database-side events that are unrelated to the SQL being emitted from the Python code.

the psycopg2 driver has an API that you can hook into PostgreSQL's LISTEN/NOTIFY system documented at https://www.psycopg.org/docs/advanced.html#asynchronous-notifications which may be useful to what you are trying to accomplish.   you would use a raw psycopg2 connection/cursor for that (see https://docs.sqlalchemy.org/en/14/core/connections.html#working-with-the-dbapi-cursor-directly)

**Reply by pankajkumar229** [reply]
Thanks very much
